src/test/java/core/basesyntax/service/impl/FileServiceCsvImplTest.java

```diff
@@ -0,0 +1,103 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.service.FileService;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FileServiceCsvImplTest {
+    private FileService fileService;
+    private String testTransactionsFileName = "src/test/resources/file.csv";
+    private String testReportFileName = "src/test/resources/report.csv";
+    private String content = "type,fruit,quantity\nb,banana,20\n";
+    private String reportString = "fruit,quantity\nbanana,20";
+    private File transactionsFile = new File(testTransactionsFileName);
+    private File reportFile = new File(testReportFileName);
+
+    @BeforeEach
+    void setUp() {
+        fileService = new FileServiceCsvImpl();
+        try {
+            transactionsFile.createNewFile();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        try {
+            Files.write(transactionsFile.toPath(), content.getBytes());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @AfterEach
+    void tearDown() {
+        transactionsFile.delete();
+        reportFile.delete();
+    }
+
+    @Test
+    void readFile_validFile_Ok() {
+        List<String> stringList = fileService.readFile(testTransactionsFileName);
+        assertEquals(2, stringList.size(), "Two lines expected");
+    }
+
+    @Test
+    void readFile_emptyFile_notOK() {
+        try {
+            Files.write(transactionsFile.toPath(), new ArrayList<>());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
not a mistake, but FYI we almost never use try-catch in tests. You can just declare that you throw exception. If it happens, the test fails automatically